### PR TITLE
[DC-1392] Refactor load_vocab script to python

### PR DESCRIPTION
**Pull request by mark-velez**
The limitations of this script will be noted in the associated ticket. Necessary adjustments will be in separate tickets.

### CODE REVIEW

File: data_steward/tools/load_vocab.py [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+        '--target_dataset_id',
+        dest='target_dataset_id',
+        action='store',
+        help=
+        'Identifies the target dataset where the vocabulary is to be loaded',
+        required=False)
+    return argument_parser
+
+
+def parse_args() -> argparse.Namespace:
```
Comment (ChaoPang) [inline data_steward/tools/load_vocab.py:229]: Just wondering, is there a particular reason for having a separate function for doing`parse_args()`?

File: data_steward/tools/load_vocab.py [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    :param dst_dataset_id: final destination to load the vocabulary in BigQuery
+    """
+    bq_client = bq.get_client(project_id=project_id)
+    sandbox_dataset_id = get_sandbox_dataset_id(dst_dataset_id)
+    sandbox_dataset = bq.define_dataset(
+        project_id,
+        sandbox_dataset_id,
+        f'Vocabulary loaded from gs://{bucket_name}',
+        label_or_tag={'type': 'vocabulary'})
+    sandbox_dataset = bq_client.create_dataset(sandbox_dataset, exists_ok=True)
```
Comment (ChaoPang) [inline data_steward/tools/load_vocab.py:178]: Can we not just do the following? 

```suggestion
    sandbox_dataset = bq.create_dataset(
        project_id,
        sandbox_dataset_id,
        f'Vocabulary loaded from gs://{bucket_name}',
        label_or_tag={'type': 'vocabulary'},
        overwrite_existing=True)
```

File: data_steward/tools/load_vocab.py [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+def get_release_date(release_date: datetime.date = None) -> str:
+    """
+    Get the name of a vocabulary release based on date
+
+    :param release_date: date the vocabulary is released
+    :return: name of vocabulary release
+    """
+    if not release_date:
+        release_date = datetime.date.today()
+    release_date_str = release_date.strftime("%Y%m%D")
```
Comment (ChaoPang) [inline data_steward/tools/load_vocab.py:246]: %D gives me a string representation '20210202/09/21' for today

```suggestion
    release_date_str = release_date.strftime("%Y%m%d")
```

File: data_steward/tools/load_vocab.py [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+    """
+    Stage files from a bucket to a dataset
+
+    :param dst_dataset: reference to destination dataset object
+    :param bq_client: a BigQuery client object
+    :param bucket_name: the location in GCS containing the vocabulary files
+    :return: list of completed load jobs
+    """
+    gcs_client = storage.Client()
+    blobs = list(gcs_client.list_blobs(bucket_name))
```
Comment (ChaoPang) [inline data_steward/tools/load_vocab.py:89]: Just wondering what if the provided `bucket_name` doesn't exist, is it okay to let it fail? The error message I got was something like "you don't have access to this bucket"

File: tests/unit_tests/data_steward/tools/load_vocab_test.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    def setUp(self):
+        project_id = 'fake_project_id'
+        dataset_id = 'fake_dataset_id'
+        bucket_name = 'fake_bucket'
+        dataset_ref = DatasetReference(project_id, dataset_id)
+        self.dst_dataset = Dataset(dataset_ref)
+        self.bq_client = Client(project_id)
+        self.bucket_name = bucket_name
+
+    def test_load_stage(self):
```
Comment (ChaoPang) [inline tests/unit_tests/data_steward/tools/load_vocab_test.py:28]: Unit test failed on CircleCI and also failed locally for me

File: tests/unit_tests/data_steward/tools/load_vocab_test.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+import mock
+from google.cloud.bigquery import Dataset, DatasetReference, Client
+from google.cloud.storage import Blob
+
+import common
+from tools import load_vocab
+
+
+class LoadVocabTest(unittest.TestCase):
```
Comment (ChaoPang) [inline tests/unit_tests/data_steward/tools/load_vocab_test.py:11]: Another main function `load` is not tested

**Review by ChaoPang — CHANGES_REQUESTED** [review]
I left some minor comments in `load_vocab.py`. `load_vocab_test.py` failed on CircleCi and also locally for me, another main function `load` is not currently tested, please add another test for it.

**Review by ChaoPang — APPROVED** [review]
Approved with the knowledge on the limitations and follow-up tickets will be created.

File: data_steward/tools/load_vocab.py (hunk 1 above)
Comment (mark-velez) [reply data_steward/tools/load_vocab.py]: Nope. Removed it.

File: data_steward/tools/load_vocab.py (hunk 2 above)
Comment (mark-velez) [reply data_steward/tools/load_vocab.py]: I did this; `create_dataset` probably didn't exist when this was written.

File: data_steward/tools/load_vocab.py (hunk 3 above)
Comment (mark-velez) [reply data_steward/tools/load_vocab.py]: You're right- fixed it.

File: data_steward/tools/load_vocab.py (hunk 4 above)
Comment (mark-velez) [reply data_steward/tools/load_vocab.py:89]: I get 
```
404 GET https://storage.googleapis.com/storage/v1/b/some-bucket-that-does-not-exist/o?projection=noAcl: Not Found
```
which I think is the appropriate way to fail

File: tests/unit_tests/data_steward/tools/load_vocab_test.py (hunk 6 above)
Comment (mark-velez) [reply tests/unit_tests/data_steward/tools/load_vocab_test.py:12]: You're right- this should be acceptance criteria for the tickets created

File: tests/unit_tests/data_steward/tools/load_vocab_test.py (hunk 5 above)
Comment (mark-velez) [reply tests/unit_tests/data_steward/tools/load_vocab_test.py:35]: Looks like this was never fully implemented. I refactored the test so it checks that expected calls to load_table_from_uri are made. Let me know if it looks ok.

**Comment by mark-velez** [comment]
@ChaoPang any suggestions on how I could get the unit tests to pass (without failing due to missing `GOOGLE_APPLICATION_CREDENTIALS`) while still mocking the results of `list_blobs` and `load_table_from_uri` as in [test_load_stage](https://github.com/all-of-us/curation/pull/733/files#diff-9852deee1060ccb7c20728bd48c125b1c341d955e018ac9d265449e6213faaa3R41-R48)?

**Pull request merged**